x-pack/plugins/alerting/common/rule.ts

```diff
@@ -16,6 +16,9 @@ import { RuleSnooze } from './rule_snooze_type';
 export type RuleTypeState = Record<string, unknown>;
 export type RuleTypeParams = Record<string, unknown>;
 
+// rule type defined alert fields to persist in alerts index
+export type RuleAlertData = Record<string, unknown>;
```

Comment:
This is a new type generic that allows rule types to define the schema of their rule type specific fields. For example, if they set `RuleAlertData` to

```
{
  foo: string;
  bar: number;
}
```

We type their alert documents using `Alert & RuleAlertData`